api/client/samples/analysis_kits/stocks_to_use_price_model.py

```diff
@@ -0,0 +1,287 @@
+"""Sample API script for modeling prices based on the stocks-to-use ratio
+
+This script creates a basic CME price valuation model for corn and soybeans
+using the US stocks-to-use ratio. 
+
+The stocks-to-use ratio is calculated each month as reported in the USDA's PS&D
+database. The price variable is defined as the monthly average 'new crop' 
+(December futures for corn and November futures for soybeans) price for the 
+relevant report month. For example, the corn-June scatterplot will compare the US
+stocks-to-use ratio from all June reports relative to the corresponding monthly 
+average December corn futures price.
+
+The output of this script will be a PDF file for each modeled crop, saved in
+the current working directory. Each PDF will have 13 pages in total, 
+including a scatterplot for each report month as well as a 
+scatterplot summarizing the entire sample.
+
+"""
+
+import os
+import calendar
+
+import pandas as pd
+
+import matplotlib.pyplot as plt
+from matplotlib.backends.backend_pdf import PdfPages
+
+from sklearn import linear_model
+
+from api.client.gro_client import GroClient
+
+API_HOST = 'api.gro-intelligence.com'
+TOKEN    = os.environ['GROAPI_TOKEN']
+client   = GroClient(API_HOST, TOKEN)
+
+ref = {'corn':{'id':274, 'futures_contract_month':12},
+       'soybeans':{'id':270, 'futures_contract_month':11}}
+
+def contract_month_history(crop, contract_month):
+    """full history of settlement prices for a specific commodity
+    and month of expiry. e.g. December corn
+
+    Parameters
+    ----------
+    crop : string
+        'corn' or 'soybeans'
+    contract_month : integer
+
+    Returns
+    -------
+    pandas.Series
+    """
+
+    market = ref[crop.lower()]['id']
+    
+    client = GroClient(API_HOST, TOKEN)
```

Comment:
updated to use a global client object. I added "client.get_df().copy()" because get_df was throwing an error after I converted reporting_date column to datetime and tried to add another series.

ValueError: You are trying to merge on datetime64[ns] and object columns. If you wish to proceed you should use pd.concat

Reply:
ok now the GroClient.get_df() now returns a df with reporting_date col cast to datetime

Reply:
cleaned up this code to use reporting_date from client's get_df